packages/flutter/test/material/icon_button_test.dart

```diff
@@ -176,6 +176,27 @@ void main() {
     expect(box.size, const Size(48.0, 600.0));
   });
 
+  testWidgets('Get size from iconThemeData', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      wrap(
+        child: IconTheme(
+          data: const IconThemeData(size: 50),
+          child: IconButton(
+            onPressed: mockOnPressedFunction.handler,
+            icon: const Icon(Icons.link),
+            constraints: const BoxConstraints(),
+          ),
+        ),
+      ),
+    );
+
+    final RenderBox iconButton = tester.renderObject(find.byType(IconButton));
```

Comment:
IMO, you should test if the icon has achieved the desired size, instead of checking for the IconButton. You can refer the tests I made: [`icon_button_test.dart`](https://github.com/flutter/flutter/pull/87643/files).

Reply:
All the pre-existing tests in `icon_button_test.dart` check for the IconButton. So I just copied it.